File: iOS/meApp/Domain/Models/API/Goal/GoalResponse.swift
import Foundation

/// Represents a response containing goal information for a user.
struct GoalResponse: Codable {
    /// The type of goal
    let type: GoalType
    /// The target goal weight
    let goalWeight: Double
    /// The date and time when the goal was created
    let createdAt: String
    /// The initial weight of the user
    let initialWeight: Double
}

File: Android/app/src/main/java/com/greatergoods/meapp/domain/model/api/goal/GoalResponse.kt
package com.greatergoods.meapp.domain.model.api.goal

/**
 * Response model for goal API operations.
 * Represents the goal data returned from the server.
 */
data class GoalResponse(
    /** The target weight for the goal */
    val goalWeight: Double,

    /** The initial weight when the goal was set */
    val initialWeight: Double,

    /** The type of goal: 'lose', 'gain', or 'maintain' */
    val type: String,

    /** Legacy field for backward compatibility */
    val goalType: String?,

    /** Whether the previous goal was met */
    val metPreviousGoal: Boolean? = null,

    /** Timestamp when the goal was created */
    val createdAt: String? = null,

    /** Timestamp when the goal was last updated */
    val updatedAt: String? = null
)
